### ISSUE TITLE
Feature: Auto completion UX

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/binderhub#588]
I imagine a user experience where you enter a git repo as a first step, and then use fetched information about the repo's branches and files to help with the selection of branch and startup file.

This experience can be compared to having to go to the repo manually to learn what the exact names of the available branches were before choosing something else than the master branch, or similarly having to look up the exact name of the files in the repo.

**Comment by choldgraf** [comment]
in this case, you're imaging someone who went to `mybinder.org` and then typed in a repo, as opposed to clicking a binder link, yeah?

**Comment by consideRatio** [comment]
@choldgraf yepp exactly - The main benefit as I see it would be for newcomers. To reduce mental effort required and reduce mistakes causing additional "lets see if it works now" verification runs.

**Comment by choldgraf** [comment]
yeah I agree there's a ton of UX improvements that could be made :-)
